src/backend/distributed/metadata/metadata_sync.c

```diff
@@ -3245,6 +3277,13 @@ citus_internal_add_shard_metadata(PG_FUNCTION_ARGS)
 		shardMaxValue = PG_GETARG_TEXT_P(4);
 	}
 
+	bool needsSeparateNode = false;
+	if (expectNeedsSeparateNode)
+	{
+		PG_ENSURE_ARGNOTNULL(5, "needs separate node");
+		needsSeparateNode = PG_GETARG_BOOL(5);
+	}
```

Comment:
> Also this is only applicable when the workers run newer versions compared to the coordinator no? If the coordinator runs a newer version compared to the workers it will start sending the new parameter to the workers, which can't map that to a valid function.

> I wonder why we want to support metadata syncing in a cluster running a non-homogenius citus version. Is there some prior art to metadata syncing on different Citus versions?

We do that to support rolling upgrades to some degree, where while the coordinator and some workers have Citus version X, some other workers have Citus version Y, where Y > X.

> if we can solve this simply with a default parameter on the sql function.

This seems like a much better option, let me try this one 👍

Reply:
Indeed we upgrade all at once in our managed service but we still put some efforts to support those kind of rolling upgrades.